### PR TITLE
BCMOHAD-21360 - Referral-Phone number Validation

### PR DESCRIPTION
**Pull request by lilianenassar-bib**
# Description

Referral-Phone number Validation

Fixes # BCMOHAD-21360

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [X ] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream moduleX

### CODE REVIEW

File: MAiD - Case Management App/force-app/main/default/objects/Referral__c/validationRules/Phone_number_Validation.validationRule-meta.xml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<ValidationRule xmlns="http://soap.sforce.com/2006/04/metadata">
+    <fullName>Phone_number_Validation</fullName>
+    <active>true</active>
+    <errorConditionFormula>OR(AND(Individual_Cell_Phone_Number__c!=null,OR(NOT(ISNUMBER(Individual_Cell_Phone_Number__c)),LEN(Individual_Cell_Phone_Number__c)!=10)),
```
Comment (cgijeffolsen) [inline MAiD - Case Management App/force-app/main/default/objects/Referral__c/validationRules/Phone_number_Validation.validationRule-meta.xml:5]: Salesforce autoformats phone numbers on entry from 6045211234 to (604) 521-1234 and this is why non-numerical characters are allowed. Test this after auto-formatting as you may need a REGEX that approves (999) 999-9999 which is length 14. @lilianenassar-bib

File: MAiD - Case Management App/force-app/main/default/objects/Referral__c/validationRules/Phone_number_Validation.validationRule-meta.xml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<ValidationRule xmlns="http://soap.sforce.com/2006/04/metadata">
+    <fullName>Phone_number_Validation</fullName>
+    <active>true</active>
+    <errorConditionFormula>OR(AND(Individual_Cell_Phone_Number__c!=null,
+NOT(REGEX(Individual_Cell_Phone_Number__c,&quot;(\\D?[0-9]{3}\\D?)[\\s][0-9]{3}-[0-9]{4}&quot;)),NOT(REGEX(Individual_Cell_Phone_Number__c,&quot;^[0-9]{10}$&quot;))),
```
Comment (cgijeffolsen) [inline MAiD - Case Management App/force-app/main/default/objects/Referral__c/validationRules/Phone_number_Validation.validationRule-meta.xml:6]: Review and testing indicates that an additional phone number is present in the interface and needs to be added: Individual_Home_Phone_Number__c @lilianenassar-bib Thank you.

**Review by cgijeffolsen — APPROVED** [review]
Approved, thank you @lilianenassar-bib 
cc @NataliaNikishina

**Review by NataliaNikishina — APPROVED** [review]
@lilianenassar-bib approved

**Pull request merged**